pkg/interface/soto/src/js/api.js

```diff
@@ -0,0 +1,49 @@
+import React from 'react';
+import ReactDOM from 'react-dom';
+import _ from 'lodash';
+
+class UrbitApi {
+  setAuthTokens(authTokens) {
+    this.authTokens = authTokens;
+    this.bindPaths = [];
+  }
+
+  bind(path, method, ship = this.authTokens.ship, appl = "soto", success, fail) {
+    this.bindPaths = _.uniq([...this.bindPaths, path]);
+
+    window.subscriptionId = window.urb.subscribe(ship, appl, path, 
+      (err) => {
+        fail(err);
+      },
+      (event) => {
+        success({
+          data: event,
+          from: {
+            ship,
+            path
+          }
+        });
+      },
+      (err) => {
+        fail(err);
+      });
+  }
+
+  soto(data) {
+    this.action("soto", "sole-action", data);
+  }
```

Comment:
You probably want to send “json” type instead of “sole-action” type so that you skip Ford mark conversions. 10x speedup.

Reply:
See also #1946. Changing this should probably be accompanied by a `//TODO revert after #1946`.